packages/core/src/actions/smartAccount/signUserOperation.ts

```diff
@@ -16,11 +17,15 @@ export const signUserOperation: <
   TChain extends Chain | undefined = Chain | undefined,
   TAccount extends SmartContractAccount | undefined =
     | SmartContractAccount
-    | undefined
+    | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 >(
   client: Client<TTransport, TChain, TAccount>,
-  args: SignUserOperationParameters<TAccount>
-) => Promise<UserOperationRequest> = async (client, args) => {
+  args: SignUserOperationParameters<TAccount, TEntryPointVersion>
```

Comment:
I don't think signer user operation should know about the Entrypoint version. if the account has the entrypoint definition attached to it, then shouldn't it be enough to just use `account.entrypoint.blah`?